### DISCUSSION TITLE
help doesnt show the default for optionals

### DISCUSSION THREAD

**Discussion by euri10** [DanCardin/cappa#124] [Ideas]
```
from dataclasses import dataclass
import cappa
from typing_extensions import Annotated


@dataclass
class Example:
    single_option: Annotated[int, cappa.Arg(short=True, help="A number")] = 42


args: Example = cappa.parse(Example, backend=cappa.backend)
print(args)
```

will produce this help:
```
❯ python ex.py --help
Usage: example [-s SINGLE_OPTION] [-h] [--completion COMPLETION]
                                                                                                                         
  Options                                                                                                                
    [-s SINGLE_OPTION]         A number                                                           
                                                                                                                         
  Help                                                                                                                   
    [-h, --help]               Show this message and exit.                                                      
    [--completion COMPLETION]  Use `--completion generate` to print shell-specific completion source. Valid options:
                               generate, complete.
```

however I find it would be definitely clearer from a user's perspective to have somehting like:
```
❯ python ex.py --help
Usage: example [-s SINGLE_OPTION] [-h] [--completion COMPLETION]
                                                                                                                         
  Options                                                                                                                
    [-s SINGLE_OPTION]         Default: 42 . A number                                                           
                                                                                                                         
  Help                                                                                                                   
    [-h, --help]               Show this message and exit.                                                      
    [--completion COMPLETION]  Use `--completion generate` to print shell-specific completion source. Valid options:
                               generate, complete.
```

wdyt ?

**Comment by DanCardin** [comment]
Yea, I think I perhaps had this originally then decided formatting-wise, it was hard to predict the formatting/casing of the users' help text such that it **always** looks consistent.

I could certainly add a `default_help=True` (or something like that) option to Arg, or I could supply `default` as a help text formatter like `help="Default: {default}. A number"`, allowing the user to essentially decide their own default formatting. I assume the goal here is **mostly** to avoid duplicating the actual value, and less about saving the help text characters?

**Reply by euri10** [reply]
I like both options except in both cases I'd have to add it for each and every Arg.

Imho showing the default should be the default ( :trollface: ) and you could have a Arg option not to show it if you dont like the formatting, kind of the opposite, but honestly just a way to show it would be awesome

**Reply by DanCardin** [reply]
well once it's a per-arg option at all, there could be a `cappa.parse/invoke`-level global option also.

Perhaps... `cappa.invoke(help_options=HelpOptions(arg_format="{help} (Default: {default})"))` or something to that effect. I very much like the idea of format strings for helping to control this, but because of the spaces, i doesn't lend itself ideally to `help` being potentially an empty string.

**Comment by DanCardin** [comment]
https://github.com/DanCardin/cappa/pull/126, lmk how you think that looks, before i commit to merging 😛. Although, i also **did** update the default to including...the default value, so for practical purposes, the details i guess might not matter to you :P
